Source/Orts.Simulation/Simulation/Signalling/Signals.cs

```diff
@@ -12116,6 +12118,7 @@ public bool CheckTimingTrigger(int reqTiming, string dumpfile)
 
         public bool TrainHasCallOn(bool allowOnNonePlatform, bool allowAdvancedSignal, string dumpfile)
         {
+            CallOnEnabled = true;
```

Comment:
Here's a difference between this code and the one presented in #155 - in the previous PR, you set this property in SIGSCRfile.cs, but now, you've made it always true. Can you explain this change?

Reply:
Rob created two additional callon functions between the previous PR and this one, and all of them call this function. Setting this variable within TrainHasCallOn() is better than having to set it separately for each function.

The aim of this variable is to show the "Enable Call On" option in the signal's context menu only for signals which support this feature.

Reply:
The purpose of the function is to "Test if train has call-on set," which strongly implies that there shouldn't be any side effects. Therefore, I think it's best to set this variable back in SIGSCRfile.cs, regardless of the code duplication.

Reply:
I moved it back to SIGSCRfile.cs